### DISCUSSION TITLE
Chord selection function

### DISCUSSION THREAD

**Discussion by cryotato** [tttapa/Control-Surface#827] [General]
im making a guitar like instrument where you select the chord but it only plays the notes of the chord when you strum,, any good pointers?

**Comment by cryotato** [comment]
I see some kind of chord functionality in the documentation but im not sure how it works and it doesnt seem to have any examples attached... right now im manually making if functions for every chord and the code is getting super long lol

**Comment by tttapa** [comment]
You can use something like this:

```cpp
#include <Control_Surface.h>

template <class Sender> void playChord(
  Sender &sender,
  MIDIAddress note,
  const IChord &chord,
  uint8_t velocity
) {
  sender.sendNoteOn(note, velocity);
  for (int8_t offset : chord)
    sender.sendNoteOn(note + offset, velocity);
}

USBMIDI_Interface midi;

void setup() {
  midi.begin();
}

void loop() {
  static Timer<millis> timer {1000};
  static uint8_t velocity = 0;
  if (timer) {
    velocity = 0x7F - velocity;
    playChord(midi, MIDI_Notes::C(3), Chords::DominantSeventh, velocity);
  }
}
```
<https://tttapa.github.io/Control-Surface-doc/Doxygen/df/d20/namespaceChords.html>

**Reply by cryotato** [reply]
wow i think this was exactly what i was looking for :) is there a way to code in your own intervals or something? ive been coding everything in manually before realizing how many years it was going to take

**Reply by tttapa** [reply]
> is there a way to code in your own intervals or something?

Yes, you can use the same syntax as the predefined chords: https://tttapa.github.io/Control-Surface-doc/Doxygen/d9/dac/Chords_8hpp_source.html#l00043
